pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,245 @@
+	case apiconfigv1.DefaultUpdateDefaultReaction:
+		desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = apiconfigv1.DefaultNodeMonitorGracePeriod.String()
+	case apiconfigv1.MediumUpdateAverageReaction:
+		desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = apiconfigv1.MediumNodeMonitorGracePeriod.String()
+	case apiconfigv1.LowUpdateSlowReaction:
+		desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = apiconfigv1.LowNodeMonitorGracePeriod.String()
+	}
+
+	_, operatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+
+	// Collect the unique set of revisions of the node static pods
+	revisionMap := map[int32]struct{}{}
+	uniqueRevisions := []int32{}
+	for _, nodeStatus := range operatorStatus.NodeStatuses {
+		revision := nodeStatus.CurrentRevision
+		if _, ok := revisionMap[revision]; !ok {
+			revisionMap[revision] = struct{}{}
+			uniqueRevisions = append(uniqueRevisions, revision)
+		}
+	}
+
+	// For each revision, check that the configmap for that revision have
+	// correct argument values or not
+	revisionsHaveSynced := true
+	for _, revision := range uniqueRevisions {
+		configMapNameWithRevision := fmt.Sprintf("%s-%d", controllerManagerConfigMapName, revision)
+		configMap, err := c.configMapClient.ConfigMaps(operatorclient.TargetNamespace).Get(ctx, configMapNameWithRevision, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		match, err := configMatchesControllerManagerArguments(configMap, desiredControllerManagerArgumentVals)
+		if err != nil {
+			return err
+		}
+		if !match {
+			revisionsHaveSynced = false
+			break
+		}
+	}
+
+	if revisionsHaveSynced {
+		// Controller Manager has Completed rollout
+		completedCondition.Status = metav1.ConditionTrue
+		completedCondition.Message = "all kube-controller-manager(s) have updated latency profile"
+		completedCondition.Reason = reasonLatencyProfileUpdated
+
+		// Controller Manager is not Progressing rollout
+		progressingCondition.Status = metav1.ConditionFalse
+		progressingCondition.Reason = reasonLatencyProfileUpdated
+
+		// Controller Manager is not Degraded
+		degradedCondition.Status = metav1.ConditionFalse
+		degradedCondition.Reason = reasonLatencyProfileUpdated
+	} else {
+		// Controller Manager has not Completed rollout
+		completedCondition.Status = metav1.ConditionFalse
+		completedCondition.Reason = reasonLatencyProfileUpdateTriggered
+
+		// Controller Manager is Progressing rollout
+		progressingCondition.Status = metav1.ConditionTrue
+		progressingCondition.Message = "kube-controller-manager(s) are updating latency profile"
+		progressingCondition.Reason = reasonLatencyProfileUpdateTriggered
+
+		// Controller Manager is not Degraded
+		degradedCondition.Status = metav1.ConditionFalse
+		degradedCondition.Reason = reasonLatencyProfileUpdateTriggered
+	}
+	_, err = c.updateConfigNodeStatus(ctx, degradedCondition, progressingCondition, completedCondition)
```

Comment:
I'm a bit worried that shuffling these conditions throughout this entire function is a bit verbose. I'd probably prefer creating full condition at every place where you're planning to update. This way you'll have a full control on what's being set at any given place. Currently I need to look through entire function and trace what each if does to the condition. Check https://github.com/openshift/library-go/blob/master/pkg/operator/v1helpers/helpers.go which I see you've looked at for some of the update functions. Now compare the usage with this: https://github.com/openshift/library-go/blob/5e6227194fbeb42769809e7026625dd1190d4ab4/pkg/operator/staticresourcecontroller/static_resource_controller.go#L341-L365 where the condition is created directly before the update invocation limiting the amount of the code I need to look back.

Reply:
https://github.com/openshift/cluster-kube-controller-manager-operator/pull/611#discussion_r871979638